### DISCUSSION TITLE
Passing variable as argument for splitFasta()

### DISCUSSION THREAD

**Discussion by SieglindeCoppens** [nextflow-io/nextflow#3388] [Q&A] [answered]
Hi!

I am facing a similar issue as #1702, but for a the splitFasta() operator.  

I wish to split a multifasta in 12 pieces that each contain roughly the same amount of sequences. Eg. if I have a multifasta with 184 sequences I want to split this into 12 files with each 16 sequences. 

I first tried to do this using this code:
```
blocksize = full_blast_mf.countFasta().map { (it + 12 - 1)/12 }.toInteger()
full_blast_mf
        .splitFasta(by: blocksize, file:true)
        .set { ch_fasta }
```
but this caused the issue:
**Value 'DataflowVariable(value=null)' cannot be used in in parameter 'by' for operator 'Fasta' -- Value don't match: class java.lang.Integer**

I have tried different solutions mentioned in #1702. Both .val and .getValue() make my workflow freeze. 

[FriederikeHanssen](https://github.com/FriederikeHanssen) mentioned that she fixed the issue for groupTuple by using the map operator (https://github.com/nextflow-io/nextflow/issues/1702#issuecomment-871182978), I tried to do the same for splitFasta().

```
full_blast_mf.countFasta().map { (it + 12 - 1)/12 }.toInteger().map{ num_fasta = it }

full_blast_mf
      .splitFasta(by: num_fasta, file:true)
      .set { ch_fasta }
```
This code runs without issues but instead of splitting in 12 pieces it splits into one file for each sequence, even though num_fasta seems to contain the right value. 

Is there some kind of workaround for this? Or should I handle this in a different way? 

Thank you!

**Comment by pditommaso** [comment]
Unfortunately, this cannot be done. Operator arguments cannot be dataflow variables

**Reply by SieglindeCoppens** [reply]
Thank you for the quick response! 

In groovy a DataflowVariable has the '.val' property. Calling this property  in nextflow causes the workflow to freeze without an error. Is this a bug? Will we be able to use the 'val' property in the future? 

Is there no way to capture the channel output in an integer?

**Comment by jemunro** [comment, marked as answer]
As an alternative you could add a process to split the FASTA file.

E.g. using https://bioinf.shenwei.me/seqkit/usage/#split
